legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabReviewApi.java

```diff
@@ -644,6 +655,54 @@ public ReviewUpdateStatus updateReview(String projectId, String reviewId)
         return getReviewUpdateStatus(gitLabProjectId, gitLabApi, rebaseMergeRequest);
     }
 
+    @Override
+    public Review editReview(String projectId, String reviewId, String title, String description, Set<String> labels)
+    {
+        LegendSDLCServerException.validateNonNull(projectId, "projectId may not be null");
+        LegendSDLCServerException.validateNonNull(reviewId, "reviewId may not be null");
+        LegendSDLCServerException.validateNonNull(title, "title may not be null");
+        LegendSDLCServerException.validateNonNull(description, "description may not be null");
+
+        GitLabProjectId gitLabProjectId = parseProjectId(projectId);
+        GitLabApi gitLabApi = getGitLabApi(gitLabProjectId.getGitLabMode());
+        MergeRequestApi mergeRequestApi = gitLabApi.getMergeRequestApi();
+
+        MergeRequest mergeRequest = checkOpenedReview(mergeRequestApi, gitLabProjectId, reviewId);
+        try
+        {
+            MergeRequestParams mergeRequestParams  = new MergeRequestParams().withTitle(title).withDescription(description);
+            if (labels != null && !labels.isEmpty())
+            {
+                mergeRequestParams.withLabels(new ArrayList<>(labels));
+            }
+
+            MergeRequest editedRequest = mergeRequestApi.updateMergeRequest(gitLabProjectId.getGitLabId(),
+                    mergeRequest.getIid(),
+                    mergeRequestParams
+            );
+            return fromGitLabMergeRequest(projectId, editedRequest);
+        }
+        catch (Exception e)
+        {
+            throw buildException(e,
+                    () -> "User " + getCurrentUser() + " is not allowed to edit review " + reviewId + " in project " + projectId,
+                    () -> "Unknown review in project " + projectId + ": " + reviewId,
+                    () -> "Error edit review " + reviewId + " in project " + projectId);
+
+        }
+    }
+
+
+    private MergeRequest checkOpenedReview(MergeRequestApi mergeRequestApi, GitLabProjectId gitLabProjectId, String reviewId)
+    {
+        MergeRequest mergeRequest = getReviewMergeRequest(mergeRequestApi, gitLabProjectId, reviewId);
+        if (!isOpen(mergeRequest))
+        {
+            throw new LegendSDLCServerException("Only open reviews can be edited: state of review " + mergeRequest.getIid() + " in project " + gitLabProjectId.toString() + " is " + getReviewState(mergeRequest), Status.CONFLICT);
+        }
+        return mergeRequest;
+    }
```

Comment:
You're calling this method from two different places (updateReview and editReview), but the error message here is only appropriate for one of those places (editReview). It's not obvious to me that it's worth abstracting this validation out into its own method. The logic of the validation is pretty trivial.